include/dlr.h

```diff
@@ -223,7 +223,7 @@ int GetDLRWeightName(DLRModelHandle* handle, int index,
  */
 DLR_DLL
 int SetDLRInput(DLRModelHandle* handle, const char* name, const int64_t* shape,
-                void* input, int dim);
+                const void* input, int dim);
```

Comment:
We get away only because `dlsym` returns a `void*`

What this PR does is less risky, since losing the `const`-ness is what's usually flagged by compilers.
However, it is important to ensure the backward compat of a stored type info is not broken, say, I have the following in my existing DLR client application code in `C++`.

```C++
// dlrHandle = dlopen();
using foo_ptr_t = int (*)(void*, const char*, const int64_t*, void*, int);
auto my_fptr = dlsym(dlrHandle, "SetDLRInput");

// my_set_in_ptr is void *
((foo_ptr_t)my_fptr)(...);  // -> does this throw an error or warning?
```

It's good to mention this case was tested explicitly (casting old pointer type on the new one)

Reply:
Manoj, `dlr_dlsym_test.cc` tests that old SetDLRInput (with non-const input) still works.
```
int (*SetDLRInput)(DLRModelHandle* handle, const char* name, const int64_t* shape,
                   void* input, int dim);
// New signature of SetDLRInput - input data pointer marked as const
int (*SetDLRInputC)(DLRModelHandle* handle, const char* name, const int64_t* shape,
                    const void* input, int dim);
                    
*(void**)(&SetDLRInput) = dlsym(dlr, "SetDLRInput");
*(void**)(&SetDLRInputC) = dlsym(dlr, "SetDLRInput");

float* img_data = img.data();
const float* cimg_data = img.data();

// Test that old SetDLRInput (with non-const input) still works
EXPECT_EQ(SetDLRInput(&model, input_name, shape, img_data, 4), 0);
// Test that new SetDLRInput accepts const and non-const inputs
EXPECT_EQ(SetDLRInputC(&model, input_name, shape, img_data, 4), 0);
EXPECT_EQ(SetDLRInputC(&model, input_name, shape, cimg_data, 4), 0);
```

Reply:
Thanks for quick turn around.

LGTM.